test/integration/client/dynamic_client_test.go

```diff
@@ -137,6 +137,28 @@ func TestDynamicClientWatch(t *testing.T) {
 		t.Fatalf("unexpected error creating dynamic client: %v", err)
 	}
 
+	testDynamicClientWatch(t, client, dynamicClient)
+}
+
+func TestDynamicClientWatchWithCBOR(t *testing.T) {
+	framework.EnableCBORServingAndStorageForTest(t)
+	framework.SetTestOnlyCBORClientFeatureGatesForTest(t, true, true)
+
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, framework.DefaultTestServerFlags(), framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(result.ClientConfig)
+	dynamicClientConfig := rest.CopyConfig(result.ClientConfig)
+	dynamicClientConfig.Wrap(framework.AssertRequestResponseAsCBOR(t))
```

Comment:
does this test the encoding of watch.Event, the watch.Event.Object, or both?

Reply:
Technically the event itself, in practice because Object is a RawExtension it must also be CBOR with no envelope. Should I make this test also skim and assert on the object bytes?

Reply:
I think it should?  I'm trying to remember all the raw extension details.. what exactly get's validated in this test for a watch even today?

Reply:
The test roundtripper tees the entire response to a buffer and CBOR-unmarshals it into an `interface{}`, so this is definitely checking that the event itself is CBOR. In both JSON and CBOR, the serialized Object is directly embedded in the serialized event. If there were an envelope, decoding the event to unstructured would give you an unstructured containing the envelope itself.

The test added in https://github.com/kubernetes/kubernetes/pull/128456/files#diff-919a89a2a5ca856fd41428749b39cefd3674e0be62d0c8bab22586ef62af01b6 _does_ separately CBOR-decode events and their embedded objects, but it's making the request directly with a RESTClient. I'll extend this test to check the encoding of the embedded object too to make sure there is no surprise that is specific to the dynamic client.

Reply:
> what exactly get's validated in this test for a watch even today

AIUI: "Induce a watch event by creating an object, verify that a dynamic client Watch with field selector matching the created object and resource version preceding the creation of that object observes the expected watch event, and that the event's object is the expected object." There is no consideration of encoding in the original test, so the intent of the CBOR variant is "behaves identically, verify that the response encoding was in fact CBOR".

Reply:
> I'll extend this test to check the encoding of the embedded object too to make sure there is no surprise that is specific to the dynamic client.

Pushed, the test now separately decodes an event, then the raw bytes of event.Object.